plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -406,10 +406,16 @@ abstract class MavenPublishBaseExtension @Inject constructor(
   private fun defaultJavaDocOption(javadocJar: Boolean, plainJavadocSupported: Boolean): JavadocJar {
     return if (!javadocJar) {
       JavadocJar.None()
-    } else if (project.plugins.hasPlugin("org.jetbrains.dokka") || project.plugins.hasPlugin("org.jetbrains.dokka-android")) {
+    } else if (project.plugins.hasPlugin("org.jetbrains.dokka-javadoc")) {
+      JavadocJar.Dokka("dokkaGeneratePublicationJavadoc")
+    } else if (project.plugins.hasPlugin("org.jetbrains.dokka")) {
       val dokkaTask = project.provider {
         val tasks = project.tasks.withType(DokkaTask::class.java)
-        tasks.singleOrNull()?.name ?: "dokkaHtml"
+        if (tasks.size == 0) {
+          "dokkaGeneratePublicationHtml"
+        } else {
+          tasks.singleOrNull()?.name ?: "dokkaHtml"
+        }
```

Comment:
With `V2Enabled` there will be 0 registered `DokkaTask` instances so we can use that as an indicator. When Dokka reached 2.1.0 we can make v2 mode required and simplify all of this to just  `JavadocJar.Dokka("dokkaGeneratePublicationHtml)`, like the javadoc block above.